tests/src/test/scala/common/WskTestHelpers.scala

```diff
@@ -17,18 +17,105 @@
 
 package common
 
+import java.time.Instant
+
+import org.scalatest.Matchers
+
 import scala.collection.mutable.ListBuffer
 import scala.util.Failure
 import scala.util.Try
 import scala.concurrent.duration.Duration
 import scala.concurrent.duration.DurationInt
 import scala.language.postfixOps
 
-import org.scalatest.Matchers
-
-import TestUtils._
 import spray.json._
-import java.time.Instant
+
+import TestUtils.RunResult
+import TestUtils.CONFLICT
+
+
+/**
+ * An arbitrary response of a whisk action. Includes the result as a JsObject as the
+ * structure of "result" is not defined.
+ *
+ * @param result a JSON object used to save the result of the execution of the action
+ * @param status a string used to indicate the status of the action
+ * @param success a boolean value used to indicate whether the action is executed successfully or not
+ */
+case class ActivationResponse(result: Option[JsObject], status: String, success: Boolean)
+
+object ActivationResponse extends DefaultJsonProtocol {
+    implicit val serdes = jsonFormat3(ActivationResponse.apply)
+}
+
+/**
+ * Activation record as it is returned from the OpenWhisk service.
+ *
+ * @param activationId a String to save the ID of the activation
+ * @param logs a list of String to save the logs of the activation
+ * @param response an Object of ActivationResponse to save the response of the activation
+ * @param start an Instant to save the start time of activation
+ * @param end an Instant to save the end time of activation
+ * @param duration a Long to save the duration of the activation
+ * @param cases String to save the cause of failure if the activation fails
+ * @param annotations a list of JSON objects to save the annotations of the activation
+ */
+ case class ActivationResult (
+        activationId: String,
+        logs: Option[List[String]],
+        response: ActivationResponse,
+        start: Instant,
+        end: Instant,
+        duration: Long,
+        cause: Option[String],
+        annotations: Option[List[JsObject]]) {
+
+        def getAnnotationValue(key: String): Option[JsValue] = {
+            Try {
+                val annotation = annotations.get.filter(x => x.getFields("key")(0) == JsString(key))
+                assert(annotation.size == 1) // only one annotation with this value
+                val value = annotation(0).getFields("value")
+                assert(value.size == 1)
+                value(0)
+            } toOption
+        }
+}
+
+object ActivationResult extends DefaultJsonProtocol {
+        private implicit val instantSerdes = new RootJsonFormat[Instant] {
```

Comment:
Formatting looks wrong here, did you install the scala formatting properties and formatted these files with it?